app/web/features/auth/notifications/PushNotificationPermission.tsx

```diff
@@ -0,0 +1,236 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import CustomColorSwitch from "components/CustomColorSwitch";
+import { Trans, useTranslation } from "i18n";
+import { AUTH } from "i18n/namespaces";
+import Sentry from "platform/sentry";
+import { useState } from "react";
+import {
+  getVapidPublicKey,
+  registerPushNotificationSubscription,
+} from "service/notifications";
+import { theme } from "theme";
+import { arrayBufferToBase64 } from "utils/arrayBufferToBase64";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  alert: {
+    marginBottom: theme.spacing(3),
+    marginTop: theme.spacing(2),
+  },
+  alertPaper: {
+    padding: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+    backgroundColor: "#ef9a9a",
+    color: theme.palette.error.contrastText,
+  },
+  alertPaperSpace: {
+    marginBottom: theme.spacing(2),
+  },
+  status: {
+    marginBottom: theme.spacing(2),
+  },
+  titleBox: {
+    display: "flex",
+    alignItems: "center",
+  },
+}));
+
+export default function PushNotificationPermission({
+  className,
+}: {
+  className: string;
+}) {
+  const { t } = useTranslation(AUTH);
+  const classes = useStyles();
+
+  const [permission, setPermission] = useState(Notification.permission);
```

Comment:
another question...

about this browser stored variable (in this case `Notification.permission`) does this trigger re-render when the user changes it? (just to know...)